src/cmd/ksh93/sh/subshell.c

```diff
@@ -326,7 +326,7 @@ static_fn void nv_restore(struct subshell *sp) {
         if (nv_isattr(mp, NV_MINIMAL) && !nv_isattr(np, NV_EXPORT)) flags |= NV_MINIMAL;
         if (nv_isarray(mp)) nv_putsub(mp, NULL, 0, ARRAY_SCAN);
         nofree = mp->nvfun ? mp->nvfun->nofree : 0;
-        _nv_unset(mp, NV_RDONLY | NV_CLONE);
+        _nv_unset(mp, NV_RDONLY);
```

Comment:
`NV_CLONE` defines an operation for the `nv_disc()` function. It cannot be meaningfully combined with symbols like `NV_RDONLY`. Note that it has the value `4` which is also known as `NV_LTOU`, `NV_UNSIGN`, `NV_HEXFLOAT`, and `NV_NOPRINT`. None of which make sense in this context. At first I thought this was a typo and it was supposed to be `NV_CLONED`. But looking at how that symbol is used that can't be the case. Too, using that symbol in this statement results in test failures. Also, if you look at how `_nv_unset()` is used elsewhere none of the other callers do anything remotely like this.

So while there may be some reason to combine `NV_READ` with one of the `NV_` symbols here using `NV_CLONE` is wrong.

Reply:
This is a good example for why I'm making these changes an we need to introduce more distinct types rather than just using `int` everywhere a bit field is needed.